src/resource.ts

```diff
@@ -508,3 +507,34 @@ function optionsToRequestInit(method: string, options?: GetRequestOptions | Post
 
 }
 
+class ActiveRefreshes<T = any> {
+  private readonly refreshBySerializedRequest: Map<string, Promise<State<T>>> = new Map<string, Promise<State<T>>>();
+
+  put(uri: string, options: GetRequestOptions | undefined, activeRefresh: Promise<State<T>>): void {
+    this.refreshBySerializedRequest.set(this.hash(uri, options), activeRefresh);
+  }
+
+  get(uri: string, options: GetRequestOptions | undefined): Promise<State<T>> | undefined {
+    return this.refreshBySerializedRequest.get(this.hash(uri, options));
+  }
+
+  has(uri: string, options: GetRequestOptions | undefined): boolean {
+    return !!this.get(uri, options);
+  }
+
+  remove(uri: string, options: GetRequestOptions | undefined): boolean {
+    return this.refreshBySerializedRequest.delete(this.hash(uri, options));
+  }
+
+  private hash(uri: string, options: GetRequestOptions | undefined): string {
+    if (!options) {
+      return objectHash(uri);
+    }
+    const sortedHeaders: Record<string, string> = {};
+    new Headers(options.getContentHeaders?.() || options.headers)
+      .forEach((value, key) => {
+        sortedHeaders[key] = value.split(', ').sort().toString();
```

Comment:
There is still an issue here if one key comes back on 2 iterations

Reply:
There is no problem

Some headers like `accept-language` accept multiple values, removing `.sort()`

Reply:
> There is no problem
Just to expand on that, there is no problem thanks to Headers implementation.

> Some headers like accept-language accept multiple values, removing .sort()
And the ordering of their values is important, therefore, we must not reorder header values.